sdk/tables/azure-data-tables/azure/data/tables/_entity.py

```diff
@@ -75,17 +78,49 @@ class EntityProperty(object):
     """
 
     def __init__(self,
-                 type=None,  # type: Union[str,EdmType] # pylint:disable=W0622
-                 value=None  # type: Any
-                 ):
+        value=None, # type: Any
+        type=None,  # type: Union[str,EdmType] # pylint:disable=W0622
+        #  value=None  # type: Any
+    ):
         """
         Represents an Azure Table. Returned by list_tables.
 
         :param Union[str, EdmType] type: The type of the property.
         :param Any value: The value of the property.
         """
-        self.type = type
-        self.value = value
+        # self.type = type
+        if type is not None:
+            self.value = value
+            self.type = type
+        elif isinstance(value, six.text_type):
+            try:
+                self.value = UUID(value)
```

Comment:
nit: you can pul `self.value = value` out of this if, elif, else statement and just set it specially in this try elif block